Add a setting toggle to change the clocks display behavior

MrChip53 opened the pull request:
**Changes**
Adds a setting toggle that allows the user to pick when the clock is shown.
Options are:
- Always
- In Menus
- In Video
- Never

**Issues**
#605

...shown. 🤦‍♂️

thornbill commented:
This should probably apply to the “ends at” time also.

nielsvanvelzen commented on app/src/main/java/org/jellyfin/androidtv/preference/UserPreferences.kt:
```diff
@@ -174,6 +174,11 @@ class UserPreferences(context: Context) : SharedPreferenceStore(
+		/**
+		 * When to show the clock.
+		 */
+		var showClock = Preference.enum("pref_show_clock", ClockBehavior.ALWAYS)
```
```suggestion
		var clockBehavior = Preference.enum("pref_clock_behavior", ClockBehavior.ALWAYS)
```

nielsvanvelzen commented on app/src/main/res/values/strings.xml:
```diff
@@ -418,4 +418,8 @@
     <string name="lbl_manual_login">Manual Login</string>
     <string name="lbl_playback_control_ends">Ends at %1$s</string>
     <string name="watch_count_overflow">99+</string>
+    <string name="pref_clock_display">Show Clock</string>
+    <string name="lbl_always">Always</string>
+    <string name="pref_clock_display_menus">In Menus</string>
+    <string name="pref_clock_display_video">In Video</string>
```
```suggestion
    <string name="pref_clock_display">Show clock</string>
    <string name="lbl_always">Always</string>
    <string name="pref_clock_display_menus">While browsing</string>
    <string name="pref_clock_display_video">During playback</string>
```

nielsvanvelzen commented on app/src/main/java/org/jellyfin/androidtv/ui/ClockUserView.java:
```diff
@@ -32,6 +38,33 @@ private void init(Context context) {
         this.addView(v);
+        TextClock tc = v.findViewById(R.id.clock);
+        ClockBehavior showClock = get(UserPreferences.class).get(UserPreferences.Companion.getShowClock());
+        if (showClock != ClockBehavior.ALWAYS) {
```
The combination of this if-statement and the switch-statement for all the `setVisibility` can be simplified for better readability.

nielsvanvelzen reviewed (changes requested):
- The clock still shows on the login screen
- The logo on the home screen looks weird without a clock

MrChip53 commented:
Addressed reviews and made the login clock respond to the setting. I don't know what you really mean by the logo looking weird. The user avatar? I think it looks okay but any suggestions on how it could look better let me know. Otherwise assuming it passes tests it should be good to go.

nielsvanvelzen commented:
/azp run

nielsvanvelzen commented on app/src/main/java/org/jellyfin/androidtv/ui/shared/ToolbarView.kt:
```diff
@@ -7,10 +7,17 @@ import android.view.ViewGroup
 class ToolbarView(context: Context, attrs: AttributeSet) : FrameLayout(context, attrs) {
 	init {
 		inflate(context, R.layout.view_toolbar, this)
+		val clockBehavior = get(UserPreferences::class.java)[clockBehavior]
```
```suggestion
		val clockBehavior = get(UserPreferences::class.java)[UserPreferences.Companion.clockBehavior]
```

Although the `UserPreferences.Companion.` doesn't look that great, it's easier to read since your variable has the exact same name. Could also just rename the variable.

nielsvanvelzen commented:
>  I don't know what you really mean by the logo looking weird.

I'm not sure either, looks fine to me. Past me should've been more descriptive haha.
